File: 0035_Search_Insert_Position/main.cpp
class Solution {
  public:
    int searchInsert(vector<int> &nums, int target) {
        if (nums.empty()) {
            return 0;
        }
        int l = 0, r = nums.size() - 1;
        while (l < r) {
            int mid = (l + r + 1) >> 1;
            if (nums[mid] <= target) {
                l = mid;
            } else {
                r = mid - 1;
            }
        }
        if (nums[l] < target) {
            return l + 1;
        }
        return l;
    }
};

File: 0035_Search_Insert_Position/golang/main.go
func searchInsert(nums []int, target int) int {
	if nums == nil || len(nums) == 0 {
		return 0
	}
	n := len(nums)
	l, r := 0, n-1
	for l <= r {
		mid := l + (r-l)/2
		if nums[mid] < target {
			l = mid + 1
		} else {
			r = mid - 1
		}
	}
	return l
}